placeholders_test.go

```diff
@@ -0,0 +1,132 @@
+package txtdirect
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestParsePlaceholders(t *testing.T) {
+	tests := []struct {
+		url       string
+		requested string
+		expected  string
+	}{
+		{
+			"example.com{uri}",
+			"https://example.com/?test=test",
+			"example.com/?test=test",
+		},
+		{
+			"example.com{uri}/{~test}",
+			"https://example.com/?test=test",
+			"example.com/?test=test/test",
+		},
+		{
+			"example.com{uri}/{>Test}",
+			"https://example.com/?test=test",
+			"example.com/?test=test/test-header",
+		},
+		{
+			"example.com{uri}/{?test}",
+			"https://example.com/?test=test",
+			"example.com/?test=test/test",
+		},
+		{
+			"example.com{dir}",
+			"https://example.com/directory/test",
+			"example.com/directory/",
+		},
+		{
+			"example.com/directory/{file}",
+			"https://example.com/directory/file.pdf",
+			"example.com/directory/file.pdf",
+		},
+		{
+			"example.com/?test={fragment}",
```

Comment:
You are mixing `query` string and `fragment`.
Query: ?querykey=queryvalue&anotherquerykey=anothervalue
Fragment: #uri_fragment_usually

More Details:
https://en.wikipedia.org/wiki/Query_string
https://en.wikipedia.org/wiki/Fragment_identifier